doc/doxygen/scripts/set_canonical_doxygen.py

```diff
@@ -22,26 +22,37 @@
 # the deal.ii repository you can use the script
 # contrib/utilities/set_canonical_webpages.py
 
-import glob
+import os
 
-filenames = glob.iglob('**/*html', recursive=True)
+LINK_START = ('<link rel="canonical" href="https://www.dealii.org/current'
+              + '/doxygen/')
 
-for filename in filenames:
-    file = open(filename, 'r')
-    new_text_data = str()
-    if '<link rel="canonical"' not in file.read():
-        file.seek(0)
-        for line in file.readlines():
-            # Do not add the canonical link twice
-            canonical_link_added = False
-            new_text_data += line
-            if (not canonical_link_added) and ('<head>' in line):
-                new_text_data += ('<link rel="canonical" href="https://www.dealii.org/current/doxygen/deal.II/'
-                                  + filename + '" />' + '\n')
-                canonical_link_added = True
-        file.close()
+def filename_generator():
+    for root, _, file_names in os.walk("./"):
+        for file_name in file_names:
+            if file_name.endswith(".html"):
+                if root == "./":
+                    yield root + file_name
+                else:
+                    yield root + "/" + file_name
 
+for filename in filename_generator():
+    # skip the header file
+    if filename == ".//header.html":
```

Comment:
Why do you use the character `//` ?

Reply:
this might no longer be necessary: I think this is a leftover from before I added some code above to deal with files in the starting directory of the search. I will take a look.